modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm

```diff
@@ -0,0 +1,13 @@
+/**
+ * Checks if a player meets certain conditions to exclude them from event selection.
+ */
+/proc/engaged_role_play_check(mob/living/carbon/human/player, station = TRUE, dorms = TRUE)
+	var/turf/player_turf = get_turf(player)
+	var/area/player_area = get_area(player_turf)
+	if(!is_station_level(player_turf.z) && station)
+		return TRUE
```

Comment:
This should check that the player is at the Interlink and similar places, and not anywhere off-station (including lavaland and deep space)